src/ansi-c/ansi_c_internal_additions.cpp

```diff
@@ -120,6 +122,29 @@ static std::string architecture_string(T value, const char *s)
          std::string(s) + "=" + std::to_string(value) + ";\n";
 }
 
+/// The maximum allocation size is determined by the number of bits that
+/// are left in the pointer of width \p pointer_width.
+///
+/// The offset needs to be able to represent up to the allocation_size, and
+/// down to -allocation_size, hence 2^(pointer_width - object_bits - 1)
```

Comment:
I think the description has it backwards. The allocation size is a result of how we represent the offset, and not the other way round.

Reply:
In my head - they are simply intertwined via the following relationship

```
max_alloc_size = 2^(bits_for_offset - 1)
```

That said I didn't like how the function returns max_alloc_size, but the description implies you input the max allocation size. I just found this a clearer way to think about the relationship - I'll have another go.

Reply:
Yes that makes sense. The current description looks good to me. My reasoning for why I preferred offset -> allocation size was that this is the direction of the computation in the method.